src/basic/io-util.h

```diff
@@ -15,7 +15,11 @@ int flush_fd(int fd);
 
 ssize_t loop_read(int fd, void *buf, size_t nbytes, bool do_poll);
 int loop_read_exact(int fd, void *buf, size_t nbytes, bool do_poll);
-int loop_write(int fd, const void *buf, size_t nbytes, bool do_poll);
+
+int loop_write_full(int fd, const void *buf, size_t nbytes, bool do_poll, usec_t timeout);
+static inline int loop_write(int fd, const void *buf, size_t nbytes, bool do_poll) {
```

Comment:
We should probably only expose `do_poll` to the full version as well, I think most of the time it's not required